internal/pkg/manifest/transform_test.go

```diff
@@ -0,0 +1,915 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package manifest
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/imdario/mergo"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBasicTransformer_Transformer(t *testing.T) {
+	type testBasicTransformerStruct struct {
+		PBool   *bool
+		PInt    *int
+		PInt64  *int64
+		PUint16 *uint16
+		PUint32 *uint32
+		PString *string
+		PSlice  *[]string
+		Slice   []string
+	}
+
+	testCases := map[string]struct {
+		original func(s *testBasicTransformerStruct)
+		override func(s *testBasicTransformerStruct)
+		wanted   func(s *testBasicTransformerStruct)
+	}{
+		"PBool overridden": {
```

Comment:
Can we combine them into three test cases?

1. Overridden
2. Explicitly overridden by zero values
3. Not overridden by nil